lab_2_tokenize_by_bpe/main.py

```diff
@@ -14,6 +14,15 @@ def prepare_word(
     :param end_of_word: a token that signifies the end of word
     :return: preprocessed word
     """
+    if not isinstance(raw_word, str) or not isinstance(start_of_word, str | None)\
```

Comment:
`isinstance` is not an optimal way to check if value is `None`: as it is a singleton, consider using `is` instead